stl/inc/filesystem

```diff
@@ -2489,12 +2489,18 @@ namespace filesystem {
         _NODISCARD static __std_win_error _Advance_and_reset_if_no_more_files(shared_ptr<_Dir_enum_impl>& _Ptr) {
             auto& _Impl = *_Ptr;
             __std_fs_find_data _Data;
-            const auto _Error = __std_fs_directory_iterator_advance(_Impl._Dir._Handle, &_Data);
-            if (_Error == __std_win_error::_Success) {
-                _Impl._Refresh(_Data); // can throw
-            } else if (_Error == __std_win_error::_No_more_files) {
+            auto _Error = __std_fs_directory_iterator_advance(_Impl._Dir._Handle, &_Data);

```

Comment:
Proposal:

```
do {
auto _Error = __std_fs_directory_iterator_advance(_Impl._Dir._Handle, &_Data);
if (_Error != __std_win_error::_Success) {
_Ptr.reset();
return _Error;
}
} while (_Is_dot_or_dotdot(_Data));
return __std_win_error::_Success;
```
?

Reply:
hmm yeah, it seems more  perfect and clear than mine, let me test it

Reply:
Great ! it works very fine !! Thanks :)